packages/now-cli/src/commands/build/index.ts

```diff
@@ -0,0 +1,185 @@
+import path from 'path';
+import _glob, { IOptions as GlobOptions } from 'glob';
+import fs from 'fs-extra';
+import plural from 'pluralize';
+import { promisify } from 'util';
+import npa from 'npm-package-arg';
+import {
+  detectBuilders,
+  Builder,
+  FileFsRef,
+  PackageJson
+} from '@now/build-utils';
+import runBuild from './run-build';
+import { NowContext } from '../../types';
+import createOutput, { Output } from '../../util/output';
+import { getAllProjectFiles } from '../../util/get-files';
+import { getYarnPath } from '../../util/dev/yarn-installer';
+import { installBuilders } from '../../util/dev/builder-cache';
+
+const globP = promisify(_glob);
+const glob = async (pattern: string, options: GlobOptions) => {
+  return (await globP(pattern, options)).map((p: string) =>
+    p.replace(/\\/g, '/')
+  );
+};
+
+const cwd = process.cwd();
+const outputDir = path.join(cwd, '.now');
+const workDir = path.join(outputDir, 'workDir');
+const buildersDir = path.join(outputDir, 'builders');
+const buildsOutputDir = path.join(outputDir, 'builds');
+
+async function getBuilds(
+  files: string[],
+  output: Output
+): Promise<Builder[] | null> {
+  let builds: Builder[] = [];
+  let nowJson;
+
+  try {
+    nowJson = await fs.readJson(path.join(cwd, 'now.json'));
+    const toResolve: Builder[] = [];
+    builds = (nowJson.builds || []).filter((build: Builder) => {
+      if (_glob.hasMagic(build.src)) {
+        toResolve.push(build);
+        return false;
+      }
+      return true;
+    });
+
+    for (const build of toResolve) {
+      const results = await glob(build.src, { cwd, nodir: true });
+      results.forEach((src: string) => {
+        builds.push({
+          src,
+          use: build.use,
+          config: build.config
+        });
+      });
+    }
+  } catch (err) {
+    output.debug('No now.json found, using zero-config');
+  }
+
+  if (builds.length === 0) {
+    let pkg;
+    try {
+      pkg = await fs.readJson(path.join(cwd, 'package.json'));
+    } catch (err) {
+      output.debug('No package.json found');
+    }
+    const { builders } = await detectBuilders(files, pkg);
+    builds.push(...(builders || []));
```

Comment:
This could probably be assigned since we know `builds.length === 0`
```suggestion
    builds = builders || [];
```